src/ML-POD/pair_pod.cpp

```diff
@@ -0,0 +1,370 @@
+
+void CPairPOD::init_style()
+{
+  if (force->newton_pair == 0)
+  error->all(FLERR,"Pair style POD requires newton pair on");
+
+  // need a full neighbor list
+
+  neighbor->add_request(this, NeighConst::REQ_FULL);
+}
+
+/* ----------------------------------------------------------------------
+   init for one type pair i,j and corresponding j,i
+------------------------------------------------------------------------- */
+
+double CPairPOD::init_one(int i, int j)
+{
+  if (setflag[i][j] == 0) error->all(FLERR,"All pair coeffs are not set");
+    scale[j][i] = scale[i][j];
+  return podptr->pod.rcut;
+}
+
+/* ----------------------------------------------------------------------
+   memory usage
+------------------------------------------------------------------------- */
+
+double CPairPOD::memory_usage()
+{
+  double bytes = Pair::memory_usage();
+  return bytes;
+}
+
+void *CPairPOD::extract(const char *str, int &dim)
+{
+  dim = 2;
+  if (strcmp(str,"scale") == 0) return (void *) scale;
+  return nullptr;
+}
+
+void CPairPOD::InitPairPOD(std::string pod_file, std::string coeff_file)
+{
+  podptr = new CPOD(lmp, pod_file, coeff_file);
+
+  lammpspairlist = 1;
+
+  if (coeff_file != "") {
+    TemplateMalloc(&podcoeff, podptr->pod.nd, backend);
+    TemplateMalloc(&newpodcoeff, podptr->pod.nd, backend);
+    TemplateMalloc(&energycoeff, podptr->pod.nd1234, backend);
+    TemplateMalloc(&forcecoeff, podptr->pod.nd1234, backend);
+    TemplateMalloc(&gd, podptr->pod.nd1234, backend);
+    podptr->podArrayCopy(podcoeff, podptr->pod.coeff, podptr->pod.nd);
+    podptr->podArrayCopy(newpodcoeff, podptr->pod.coeff, podptr->pod.nd);
+  }
+}
+
+
+void CPairPOD::free_tempmemory()
+{
+  TemplateFree(rij, backend);
+  TemplateFree(idxi, backend);
+  TemplateFree(ai, backend);
+  TemplateFree(aj, backend);
+  TemplateFree(ti, backend);
+  TemplateFree(tj, backend);
+  TemplateFree(numneighsum, backend);
+  TemplateFree(typeai, backend);
+  TemplateFree(tmpmem, backend);
+}
+
+void CPairPOD::free_atommemory()
+{
+  TemplateFree(forces, backend);
+  TemplateFree(stress, backend);
+  if (atommemory) {
+  TemplateFree(atomtype, backend);
+  TemplateFree(pos, backend);
+  TemplateFree(vel, backend);
+  }
+}
+
+void CPairPOD::free_memory()
+{
+  free_tempmemory();
+  free_atommemory();
+}
+
+void CPairPOD::allocate_tempmemory()
+{
+  TemplateMalloc(&rij, dim*nijmax, backend);
+  TemplateMalloc(&idxi, nijmax, backend);
+  TemplateMalloc(&ai, nijmax, backend);
+  TemplateMalloc(&aj, nijmax, backend);
+  TemplateMalloc(&ti, nijmax, backend);
+  TemplateMalloc(&tj, nijmax, backend);
+  TemplateMalloc(&numneighsum, nablockmax+1, backend);
+  TemplateMalloc(&typeai, nablockmax, backend);
+  TemplateMalloc(&tmpmem, szd, backend);
+}
+
+void CPairPOD::allocate_atommemory()
+{
+  TemplateMalloc(&forces, dim*nmaxatom, backend);
+  TemplateMalloc(&stress, 9, backend);
+  if (atommemory) {
+  TemplateMalloc(&atomtype, nmaxatom, backend);
```

Comment:
Please check for proper indentation. LAMMPS has a clang-format configuration file in the src folder. You can use that with clang-format to re-indent and reformat your files according to LAMMPS conventions.

Reply:
The code is re-indented.